python/setup.cfg

```diff
@@ -43,6 +43,7 @@ package_dir =
 packages = find:
 python_requires = >=3.7
 install_requires =
+    attrs
```

Comment:
Is there an alternative in the standard library we can use?

The fewer libraries we add as a dependency, the more we can allow users to pick their own versions and not have to worry about conflicts.

Reply:
There is Python built-in [dataclasses support](https://docs.python.org/3/library/dataclasses.html), which essentially was influenced by the [attrs community](https://www.attrs.org/en/stable/why.html) but it always seems to lack behind in terms of features. Unfortunately, pre-Python 3.10 dataclasses isn't as powerful as attrs. For instance, it doesn't support "kw_only" out of the box and requires us to write custom decorators to support that.

I did talk briefly about this with Sam in [this conversation thread](https://github.com/apache/iceberg/pull/4706#discussion_r867216868).

Reply:
While the dataclasses support may lag, we've had a lot of pushback around library creep.  I don't feel like the added value warrants bringing in a library in this case.  This may feel overly conservative, but with Iceberg being as low-level as it will be in many frameworks moving forward, we really want to keep the dependencies as slim as possible.